Applications/WebPortal/WebApp/library/source/pages/group_page/group_page.tsx

```diff
@@ -0,0 +1,217 @@
+import * as Beam from 'beam';
+import { HBoxLayout, Padding, VBoxLayout } from 'dali';
+import * as React from 'react';
+import { DisplaySize, HLine } from '../..';
+import { GroupSubPage } from './group_sub_page';
+import { MenuBar } from './menu_bar';
+
+interface Properties {
+
+  /** The group being displayed. */
+  group: Beam.DirectoryEntry;
+
+  /** The sub page currently selected. */
+  subPage: GroupSubPage;
+
+  /** Determines the layout used to render the page. */
+  displaySize: DisplaySize;
+
+  /** Indicates a sub-page menu item was clicked.
+   * @param subPage - The GroupSubPage that was clicked.
+   */
+  onMenuClick?: (subPage: GroupSubPage) => void;
+}
+
+/** Implements the container used to display group information. */
+export class GroupPage extends React.Component<Properties> {
+  public static readonly defaultProps = {
+    onMenuClick: () => {}
+  }
+
+  public render(): JSX.Element {
+    const Header = (() => {
+      switch(this.props.displaySize) {
+        case DisplaySize.LARGE:
+          return LargeHeader;
+        case DisplaySize.MEDIUM:
+          return MediumHeader;
+        case DisplaySize.SMALL:
+          return SmallHeader;
+      }
+    })();
+    return (
+      <VBoxLayout height='100%' width='100%'
+          style={{overflow: 'hidden' as 'hidden'}}>
+        <Header name={this.props.group.name}>
+          <MenuBar displaySize={this.props.displaySize}
+            selected={this.props.subPage} onClick={this.props.onMenuClick}/>
+        </Header>
+        {this.props.children}
+      </VBoxLayout>);
+  }
+}
+
+class HeaderUnderline extends React.PureComponent {
+  public render(): JSX.Element {
+    return <HLine color='#E6E6E6' height='1px'/>;
+  }
+}
+
+interface HeaderProps {
+  name: string;
+}
+
+class LargeHeader extends React.Component<HeaderProps> {
+  public render(): JSX.Element {
+    return (
+      <VBoxLayout width='100%'>
+        <HBoxLayout width='100%' height='40px'>
+          <Padding/>
+          <HBoxLayout width='1036px' height='40px'>
+            <Padding size='18px'/>
+            {this.props.children}
+            <Padding/>
+            <div style={LargeHeader.STYLES.usernameContainer}>
+              <div style={LargeHeader.STYLES.usernameWrapper}>
+                <UsernameLabel name={this.props.name} height='40px'/>
+              </div>
+            </div>
+            <Padding size='18px'/>
+          </HBoxLayout>
+          <Padding/>
+        </HBoxLayout>
+        <HeaderUnderline/>
+      </VBoxLayout>);
+  }
+
+  private static readonly STYLES = {
+    usernameWrapper: {
```

Comment:
Shouldn't this also be renamed to group name rather than username?

Reply:
Makes sense. I'll change all instances of username to groupName.